src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,142 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int DEFAULT_THRESHOLD = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private static final int GROW_COEFFICIENT = 2;
+
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size = 0;
+
+    public MyHashMap() {
+        this.threshold = DEFAULT_THRESHOLD;
+        this.table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        int bucketIndex = getBucketByKey(key);
+        if (size >= threshold) {
```

Comment:
let's check it before getting the bucket index

Reply:
fixed this